pkg/k8s/init.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package k8s abstracts all Kubernetes specific behaviour
+package k8s
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/cilium/pkg/nodeaddress"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Init initializes the Kubernetes package. It is required to call Configure()
+// beforehand.
+func Init() error {
+	if !IsEnabled() {
+		return nil
```

Comment:
Shouldn't we return an error here? Why is the user trying to Init something that is not enabled?

Reply:
It's actually a duplicate condition because `Init()` is already protected, I'll remove it.

```
        if k8s.IsEnabled() {
                if err := k8s.Init(); err != nil {
                        log.WithError(err).Fatal("Unable to initialize Kubernetes subsystem")
                }
```